modules/kozlov_m_graph_diam/include/graph.h

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 Kozlov Mikhail
+
+#ifndef MODULES_KOZLOV_M_GRAPH_DIAM_INCLUDE_GRAPH_H_
+#define MODULES_KOZLOV_M_GRAPH_DIAM_INCLUDE_GRAPH_H_
+
+
+#include <vector>
+#include <cstdint>
+#include <array>
+#include <limits>
+#include <stdexcept>
+#include <queue>
+#include <algorithm>
+#include <utility>
+#include <iostream>
+
+const uint64_t INF = std::numeric_limits<uint64_t>::max();
+
+template<uint32_t N>
+class Graph {
+ protected:
+    std::vector<std::vector<uint64_t>> dist;
+    std::vector<std::vector<std::pair<uint32_t, uint64_t>>> adj;
+    bool valid = true;
+    void find_dist(uint32_t start) {
+        std::priority_queue<std::pair<uint64_t, uint32_t>> q;
+
+        q.emplace(0, start);
+        dist[start][start] = 0ULL;
+
+        while (!q.empty()) {
+            uint32_t x = q.top().second;
+            q.pop();
+
+            for (auto e : adj[x]) {
```

Comment:
Лучше избегать лишнего копирования, если это возможно.
```suggestion
            for (const auto& e : adj[x]) {
```